ql/src/java/org/apache/hadoop/hive/ql/Context.java

```diff
@@ -774,21 +812,25 @@ public void setResDir(Path resDir) {
     resDirPaths = null;
   }
 
-  public void clear() throws IOException {
+  public void clear() throws IOException{
```

Comment:
Check for other callers of Context.clear() - seems like TezTask can call this as well? Will these other callers mess things up since they are calling with deleteResultDir=true?

Reply:
Looking into this

Reply:
Rest of the calls are from immediately after executing tasks. Since result hasn't been cached at that point it is safe for the context clear to delete the result cache files (if there are any)